packages/expo-asset/ios/AssetModule.swift

```diff
@@ -17,6 +17,9 @@ public class AssetModule: Module {
   public func definition() -> ModuleDefinition {
     Name("ExpoAsset")
 
+    // NOTE: this is exposed in JS as globalThis.expo.modules.ExpoAsset.downloadAsync
+    // and potentially consumed outside of Expo (e.g. RN vector icons)
+    // do NOT change the function signature as it'll break consumers!
```

Comment:
If we ever change a public JS API (e.g. `Font.loadAsync`) then it's clear that we need to document a deprecation or a breaking change.

But a lot of the native module apis have a thin JS layer around them and the native code can change while the JS API stays the same. An example could be when some native module `Function` is refactored to use a `Record` instead of a `Dictionary` or a list of several parameters. In most other apis that wouldn't be an issue, because the JS api is the same. But here it'd be a problem.

Additionally, this api actually isn't public so it'd be easy to assume that changing it is harmless.

Reply:
Does it make sense to just make this public? (since it will be used publicly)

Reply:
We want that api to be used without depending on `expo-font` or `expo-asset`. It'd be something like [this](https://github.com/vonovak/react-native-vector-icons/blob/d0e95bd05fea4d03f4be64683c5318aed4de2e04/packages/common/src/dynamicFontLoading.ts#L67).

Reply:
added `getLoadedFonts()` function

Reply:
I previously said the api isn't public, which isn't completely true. It's exposed via [downloadasync](https://docs.expo.dev/versions/latest/sdk/asset/#downloadasync). That is a JS api though, so technically, the native api still isn't public.
I think the comment still has value in case someone comes to refactor the native api.